s3/README.md

```diff
@@ -0,0 +1,207 @@
+| 25.811736633324347           | 1.545956750046378          | 140162          |                 |
++------------------------------+----------------------------+-----------------+-----------------+
+
+Query took: 0.015628708 seconds
+```
+
+## For private S3 bucket
+
+**LIMITATION:** Spice.ai currently only supports authentication with AWS S3 using `aws_access_key_id` and `aws_secret_access_key`. Please [submit an issue](https://github.com/spiceai/spiceai/issues/new?template=feature_request.md) if you would like to see support for other authentication methods.
+
+**Step 1.** Prepare S3 bucket
+
+- Create a new AWS S3 bucket `yourcompany-bucketname-datasets`.
+- Create a path `tax_trips` in the above bucket.
+- Download [taxi_trips dataset](https://d37ci6vzurychx.cloudfront.net/trip-data/yellow_tripdata_2024-01.parquet) parquet, and upload it into `taxi_trips` path in the bucket.
+
+**Step 2.** Prepare AWS IAM user
+
+- Create a new IAM user with the following inline policy:
+  ```json
+  {
+    "Version": "2012-10-17",
+    "Statement": [
+      {
+        "Effect": "Allow",
+        "Action": ["s3:ListBucket"],
+        "resource": "arn:aws:s3:::yourcompany-bucketname-datasets"
+      },
+      {
+        "Effect": "Allow",
+        "Action": ["s3:GetObject"],
+        "resource": "arn:aws:s3:::yourcompany-bucketname-datasets/*"
+      }
+    ]
+  }
+  ```
+- Create an access key that contains `aws_access_key_id` and `aws_secret_access_key` for the IAM user under `Security credentials` tab.
+
+**Step 3.** Log into S3 using the Spice CLI.
+
+```
+spice login s3 -k <aws_access_key_id> -s <aws_secret_access_key>
+```
+
+The following output is shown:
+
+```
+Successfully logged in to s3
+```
+
+**Step 4.** Initialize a Spice application, then start the Spice runtime.
+
+```bash
+spice init s3-demo-project
+```
+
+```bash
+cd s3-demo-project
+spice run
+```
+
+**Step 5.** Configure the dataset to connect to S3:
+
+```bash
+spice dataset configure
+```
+
+Enter the name of the dataset:
+
+```bash
+dataset name: (s3-demo-project)  taxi_trips
+```
+
+Enter the description of the dataset:
+
+```
+description: taxi trips in s3
+```
+
+Specify the location of the dataset:
+
+```bash
+from: s3://yourcompany-bucketname-datasets/taxi_trips/
+```
+
+Select "y" when prompted whether to locally accelerate the dataset:
+
+```bash
+Locally accelerate (y/n)? y
+```
+
+The following output is shown:
+
+```
+Dataset settings written to `datasets/taxi_trips/dataset.yaml`!
+```
+
+If the login credentials were entered correctly, the dataset will have loaded into the runtime. The following output is shown in the Spice runtime terminal:
+
+```
+2024-03-26T22:18:02.062394Z  INFO runtime: Loaded dataset: taxi_trips
+2024-03-26T22:18:02.062462Z  INFO runtime::dataconnector: Refreshing data for taxi_trips
+```
+
+**Step 6.** Run queries against the dataset using the Spice SQL REPL. (see Step 3 above)
```

Comment:
Just put the SQL here instead of referring to step 3 above.  There are actually 2 step 3's above :).

Reply:
god, my math teacher would be mad at me.